synapse/config/emailconfig.py

```diff
@@ -74,14 +74,75 @@ def read_config(self, config):
             "account_validity", {},
         ).get("renew_at")
 
-        if self.email_enable_notifs or account_validity_renewal_enabled:
+        self.email_enable_password_reset_from_is = email_config.get(
+            "enable_password_reset_from_is", False,
+        )
+        self.enable_password_resets = (
+            self.email_enable_password_reset_from_is
+            or (not self.email_enable_password_reset_from_is and email_config != {})
```

Comment:
Isn't this just `self.enable_password_resets = self.email_enable_password_reset_from_is or email_config != {}`?